src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -86,6 +86,8 @@ export class IndexPattern implements IIndexPattern {
   // savedObject version
   public version: string | undefined;
   public sourceFilters?: SourceFilter[];
+  public dataSourcesJSON: string | undefined;
```

Comment:
- why isn't it `dataSourceId`?
- If it has to be a JSON as the name indicated, why not use the `DataSourceType` you previously defined.

Reply:
@kristenTian any comments?

Reply:
offline synced during meeting, will add separate support for get